### DISCUSSION TITLE
HDPro - says streaming request from and IP but that is as far as it gets

### DISCUSSION THREAD

**Discussion by slusk83** [GerryDazoo/Slinger#506] [Q&A]
I have the exe running on a windows machine.  Below is the output i get when trying to open the stream on vlc.  The audio/video never starts and vlc just sets there.

Please let me know what the heck I am doing wrong.  

Version : 3.08f Running on Windows-10-10.0.17763-SP0 pid= 8752 C:\Users\Administrator\Downloads\Slinger-main\slingbox_server.exe
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams using URL slingbox.
starting up on  port 8080 8388608
Streamer Running:  10 config.ini SLINGBOX 8080 8080 8388608
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
An attempt was made to access a socket in a way forbidden by its access permissions
8080 No valid slingbox ip info found in config.ini
Finding Slingbox on local network. My IP Info =  192.168.5.122
Slingbox Found 192.168.5.90 5001 " bobcat83's Slingbox " FinderID 4ACB79C62D178849A2D3B8D7A58BA2A0
8080 Using slingbox at  ('192.168.5.90', 5001)
Streamer:  8080 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
01/24/2023, 17:06:26.048   Streaming request from ('192.168.5.120', 53924)

Slinginfo  "Solo/Pro/ProHD" 5 30 192.168.5.90 5001 3000 10
8080 Starting Stream for  192.168.5.120:53924
Connecting... ('192.168.5.90', 5001) Control
cmd: 0x67 err: 0x2b 8
8080 Error Starting Session. Check your admin password in config.ini file!
8080 ERROR: Slingbox session startup failed.
8080 Logging Out
EXCEPTION [WinError 10054] An existing connection was forcibly closed by the remote host 0x101 0 0x68 0 3 0 0x2000 0x0
Streamer:  8080 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

 
Below is my config.

[SLINGBOX]
;sbtype="350/500/M1/M2"
sbtype="Solo/Pro/ProHD"
;sbtype="240"
password=**********

;uncomment the next lines ipaddress and port with your local network info
;if you don't want the server to automatically find your 
;slingbox on the local network. Needed if server and slingbox are
;not on the same LAN segment. Also removes requirement for the netifaces module  
;ipaddress=192.168.117.110
;port=5201

;Replace with your finderid, If and only if you need remote access and don't  
;have a static ip and you plan on using my service to access you server remotely. 
;Please read the release notes. Most people don't need this.
;finderid=BAD95DAED609364C8204F2BC5FBDC363

;If you've got more than one slingbox set name. THe server will use this when generating 
;logs to make it easier to see what's going on. 
;name=something

;Following are the default values
;Default resolution when server starts. You can change this and reconnect without having to restart the server
;valid range = 1..15
Resolution=1

;Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolution
FrameRate=6

;Min 50 Max 8000
;VideoBandwidth=2000

;Min 0 (auto) max 63
VideoSmoothness=0

;Send Iframe every n seconds Max 30
;IframeRate=5

;Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32
AudioBitRate=32

;Video Source  0, 1, 2, 3 depending on your hardware corresponds to one of
;Composite, Component, S-Video, HDMI or Tuner.
;Pro:        0=(Analog)Tuner   1=Composite   2=S-Video     3=Component (Via HDMI to Component Cable)
;ProHD:      0=Tuner           1=Composite   2=S-Video     3=Component
;Solo:       0=Composite       1=S-Video     2=Component
;500:        0=Composite       1=Component   2=HDMI
;350/M1/M2:  0=Composite       1=Component
; If you don't set this, the code will use the last configured input
; I recommend setting this value. If not set correctly, you'll often only see a black screen.
; Note: Setting invalid options may cause unexpected hardware behavior. For example, setting
;  2 or 3 for M2 hardware causes it to reboot.
VideoSource = 1

[SERVER]
;local port number for the server to listen on for connections
port=8080
maxstreams=10
;enableremote=yes

[REMOTE]
;see release notes for changing this if your remote doesn't work, But as a start make it 
;the same as your configured VideoSource 0-3. See above.
code=1
;path to an external remote control definitions file, if the defaults
;aren't working for you or you want to change the format of the web page.
;use the supplied remote.txt as a starting point.
;include=remote.txt

**Comment by GerryDazoo** [comment]
You didn't post the complete server output. :-(
I'd like to know what version your running. You should have gotten an error message. So without the COMPLETE server output I can only guess what's wrong. So 
1) Bad Password
2) Bricked Box
3) Box is off line or in a wonky state (Wonky is an official newtworking term)

**Reply by lepmail** [reply]
I'm afraid it's bricked -- cmd: 0x67 err: **0x2b** 8.

**Reply by slusk83** [reply]
i just updated the above.

**Comment by slusk83** [comment]
I grabbed another spare HD Pro and going to vlan it off where it doesnt have network access.  I am also going to have a virtual that has a foot in both prod and dead vlan so I can grab the info from the slingbox and send it out the server.  Hope this works.

**Comment by GerryDazoo** [comment]
Your windows firewall/defender is blocking access.

**Comment by slusk83** [comment]
I moved to a SB200 that I had.   Below is where I got which seems considerably further but abruptly ends.

Version : 3.08f Running on Windows-10-10.0.17763-SP0 pid= 6104 C:\Users\Administrator\Downloads\Slinger-main\slingbox_server.exe
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams using URL slingbox.
Streamer Running:  10 config.ini SLINGBOX 8080 8080 8388608
starting up on  port 8080 8388608
Checking for slingbox at ('192.168.5.91', 5004)
('192.168.5.91', 5004) OK
slingbox Using slingbox at  ('192.168.5.91', 5004)
Streamer:  slingbox Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
An attempt was made to access a socket in a way forbidden by its access permissions
01/25/2023, 13:01:24.880   Streaming request from ('192.168.5.120', 57861)

Slinginfo  "240" 0 15 192.168.5.91 5004 3072 10
slingbox Starting Stream for  192.168.5.120:57861
Connecting... ('192.168.5.91', 5004) Control
cmd: 0x7e err: 0x13 176
slingbox Box in use! Kicking off other user.
slingbox Selecting Video Source 0
Keycodes= [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79]
VideoParameters: Resolution= 0 FrameRate= 15 VideoBandwidth= 2000 VideoSmoothness= 0 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.5.91', 5004) Stream
slingbox Stream started at 01/25/2023, 13:01:28.70  424 0
01/25/2023, 13:01:29.699  slingbox Stream Terminated for  192.168.5.120:57861
slingbox Shutting down connections
slingbox Logging Out
Streamer:  slingbox Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

**Comment by slusk83** [comment]
It is actually working.  Something is going on with VLC where it wouldn't work but windows media player works like a charm

**Reply by ChrizK-oldGit** [reply]
> It is actually working. Something is going on with VLC where it wouldn't work but windows media player works like a charm

Hi, I believe I am getting the same response as @slusk83 last reported in January.
I am currently using slingbox_server.exe on Windows 11 and attempting to use my mobile device with MX Player.  Both devices are attached to the network via WiFi.
Given that the attempt initiates the stream, would I be correct to assume that the Windows 11 firewall is not a concern?

@slusk83 how do you use Windows Media Player (Legacy?) to connect to the stream?  I have enable Stream (out and in), but I cannot see any method to insert a URL.

Any thoughts how I can trouble shoot?
If I was using incorrect VideoParameters, would the Stream Stop Suddenly?
(I believe my HDMI>YPbPr converter is outputting at 1920x1080P)

What is "No remote keys configured"?
I have had to reset my M1 so that I can default the admin password.
I have changed the config.ini to enableremote=no, VideoSource = 1 (VideoSource = 0 displays the same warning)

Just updated providing the initial startup, as I noticed Error: Can't find streamer for 57120 in dict_keys(['/slingbox/57120'])
Also, I created my directory under Program Files\slingbox.  Every time I edit config.ini, I have to do so as admin ... could this be a potential problem?

```
Version : 4.01 Running on Windows-10-10.0.22621-SP0 pid= 18532 C:\Program Files\slingbox\slingbox_server.exe
Using config file config.ini
Connection Manager Running on port 57120 with 10 max streams using URL slingbox.
Building page for Slingbox
Using built in default remote page definition.
Starting Streamer Thread for Slingbox
Broadcast Responder Running
Streamer Running:  10 config.ini SLINGBOX 57120 57120 8388608
starting up on  port 57120 8388608
57120 Checking for slingbox at ('', 5201) 0
Server IP address is 192.168.0.211
Server IP address is 172.25.192.1
57120 Error connecting to  ('', 5201)
57120 No valid slingbox ip info found in config.ini
57120 Finding Slingboxes on local network. My IP Info =  192.168.0.211
57120 Found at 192.168.0.15 5301 " My Slingbox " FinderID Dxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx ProductID M1
57120 Finding Slingboxes on local network. My IP Info =  172.25.192.1
57120 Using slingbox at  ('192.168.0.15', 5301)
Streamer:  57120 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
03/27/2023, 18:18:40.902   Streaming request from ('192.168.0.195', 36056)  /
Error: Can't find streamer for 57120 in dict_keys(['/slingbox/57120'])
03/27/2023, 18:18:40.969   Streaming request from ('192.168.0.195', 36066)  /slingbox
STREAM=192.168.0.195:36066:0
```

```
Slinginfo  M1 12 30 192.168.0.15 5301 3072 10 admin
57120 Starting Stream for  192.168.0.195:36066
57120 Opening Control stream 0x2000 0 0
Connecting... ('192.168.0.15', 5301) Control
57120 Selecting Video Source 1
57120 Warning: No remote keys configured, using correct VideoSource?
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.0.15', 5301) Stream
57120 Stream started at 03/27/2023, 18:18:42.815  0 0
03/27/2023, 18:18:42.816  57120 Stream Stopped Unexpectly. Kicked Off?
57120 Shutting down connections
57120 Logging Out
Streamer:  57120 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
03/27/2023, 18:18:42.872   Streaming request from ('192.168.0.195', 36068)  /slingbox
STREAM=192.168.0.195:36068:0
```
```
Slinginfo  M1 12 30 192.168.0.15 5301 3072 10 admin
57120 Starting Stream for  192.168.0.195:34010
57120 Opening Control stream 0x2000 0 0
Connecting... ('192.168.0.15', 5301) Control
57120 Selecting Video Source 0
57120 Warning: No remote keys configured, using correct VideoSource?
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.0.15', 5301) Stream
57120 Stream started at 03/27/2023, 17:53:18.70  0 0
03/27/2023, 17:53:18.70  57120 Stream Stopped Unexpectly. Kicked Off?
57120 Shutting down connections
57120 Logging Out
Streamer:  57120 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

```

**Reply by GerryDazoo** [reply]
If the stream starts it's not a firewall issue.
ProgramFiles is a protected folder in WindowsLand. Put it some place else OR add the full path to the config file to the command line to a folder that you have write permissions for. But no it's not a problem.
Video Source 0 is the composite input on the M1. You want video source = 1 Component input.
Try SlingerPlayer https://github.com/GerryDazoo/SlingerPlayer.

**Reply by ChrizK-oldGit** [reply]
Thank you for the (very quick) suggestion ... I'm afraid SlingPlayer results in exactly the same behaviour.

NB I moved to C:/slingbox with no change, I assumed video source 1 was correct,  but also tried 0 in case Audio Only would help with troubleshooting (no change).
Also tried resolution 16 (as per readme, but I think the comment in config.ini is incorrect, as it says valid range = 1..15)

Any more suggestions?

(I have been spending a lot of time reviewing the Comments, trying everything I can think of.   Appreciate where mike-uk is coming from ...perhaps I should only ask in my own [Getting started with Slingbox M1, the next steps](https://github.com/GerryDazoo/Slinger/discussions/567#top))

**Reply by GerryDazoo** [reply]
Yes the comment is in wrong, 16 is valid.
I'm not sure the M1 will eat a 1080P source. Can you reduce the input to something else 1080I or 720P?
VideoSource 1 is the correct value. 
Please post the server output when using videosource=1
Here's a sample of what I get when using my M1
03/27/2023, 16:37:21.692   Streaming request from ('192.168.117.122', 51692)  /sb/M1
STREAM=192.168.117.122:51692:0

Slinginfo  350/500 12 60 192.168.117.115 5301 3072 4 3s4z2141a121ujn
M1 Starting Stream for  192.168.117.122:51692
M1 Opening Control stream 0x2000 0 0
Connecting... ('192.168.117.115', 5301) Control
M1 Selecting Video Source 1
M1 Keycodes= [1, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22, 23, 24, 25, 26, 27, 28, 29, 31, 32, 33, 34, 35, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 53, 54, 55, 56, 57, 58, 59, 60]
VideoParameters: Resolution= 12 FrameRate= 60 VideoBandwidth= 6000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.117.115', 5301) Stream
M1 Stream started at 03/27/2023, 16:37:23.622  749 0
